### ISSUE TITLE
Enforce native SDK dependencies

### ISSUE THREAD

**Issue by sahildudeja-tide** [DataDog/dd-sdk-flutter#583]
### Stack trace

```
W/Datadog ( 3310): [_dd.sdk_core.default]: Logs feature receive an event of unsupported type=com.datadog.android.core.feature.event.JvmCrash.Logs.
W/Datadog ( 3310): [_dd.sdk_core.default]: RUM feature receive an event of unsupported type=com.datadog.android.core.feature.event.JvmCrash.Rum.
W/FirebaseCrashlytics( 3310): Unable to read App Quality Sessions session id.
E/AndroidRuntime( 3310): FATAL EXCEPTION: main
E/AndroidRuntime( 3310): Process: co.tide.tideplatform.uk.wip, PID: 3310
E/AndroidRuntime( 3310): java.lang.NoClassDefFoundError: Failed resolution of: Lcom/datadog/android/core/internal/system/DefaultBuildSdkVersionProvider;
E/AndroidRuntime( 3310): 	at com.datadog.android.rum.internal.DefaultAppStartTimeProvider.<init>(DefaultAppStartTimeProvider.kt:18)
E/AndroidRuntime( 3310): 	at com.datadog.android.rum.internal.domain.scope.RumApplicationScope.<init>(RumApplicationScope.kt:37)
E/AndroidRuntime( 3310): 	at com.datadog.android.rum.internal.monitor.DatadogRumMonitor.<init>(DatadogRumMonitor.kt:72)
E/AndroidRuntime( 3310): 	at com.datadog.android.rum.internal.monitor.DatadogRumMonitor.<init>(DatadogRumMonitor.kt:55)
E/AndroidRuntime( 3310): 	at com.datadog.android.rum.Rum.createMonitor(Rum.kt:74)
E/AndroidRuntime( 3310): 	at com.datadog.android.rum.Rum.enable(Rum.kt:64)
E/AndroidRuntime( 3310): 	at com.datadog.android.rum.Rum.enable$default(Rum.kt:35)
E/AndroidRuntime( 3310): 	at com.datadoghq.flutter.DatadogRumPlugin.enable(DatadogRumPlugin.kt:177)
E/AndroidRuntime( 3310): 	at com.datadoghq.flutter.DatadogRumPlugin.onMethodCall(DatadogRumPlugin.kt:117)
E/AndroidRuntime( 3310): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:267)
E/AndroidRuntime( 3310): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:292)
E/AndroidRuntime( 3310): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:319)
E/AndroidRuntime( 3310): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
E/AndroidRuntime( 3310): 	at android.os.Handler.handleCallback(Handler.java:958)
E/AndroidRuntime( 3310): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 3310): 	at android.os.Looper.loopOnce(Looper.java:230)
E/AndroidRuntime( 3310): 	at android.os.Looper.loop(Looper.java:319)
E/AndroidRuntime( 3310): 	at android.app.ActivityThread.main(ActivityThread.java:8913)
E/AndroidRuntime( 3310): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 3310): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
E/AndroidRuntime( 3310): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
E/AndroidRuntime( 3310): Caused by: java.lang.ClassNotFoundException: Didn't find class "com.datadog.android.core.internal.system.DefaultBuildSdkVersionProvider" on path: DexPathList[[zip file "/data/app/~~9s2yAEev-6h_HnQ40iw6QQ==/co.tide.tideplatform.uk.wip-nkCnL9du1CWtf3-EJWptUQ==/base.apk"],nativeLibraryDirectories=[/data/app/~~9s2yAEev-6h_HnQ40iw6QQ==/co.tide.tideplatform.uk.wip-nkCnL9du1CWtf3-EJWptUQ==/lib/arm64, /data/app/~~9s2yAEev-6h_HnQ40iw6QQ==/co.tide.tideplatform.uk.wip-nkCnL9du1CWtf3-EJWptUQ==/base.apk!/lib/arm64-v8a, /system/lib64, /system/system_ext/lib64]]
E/AndroidRuntime( 3310): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
E/AndroidRuntime( 3310): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
E/AndroidRuntime( 3310): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
E/AndroidRuntime( 3310): 	... 21 more
```

### Reproduction steps

Pubspec configuration:
```
dependencies:
  datadog_flutter_plugin: 2.2.0
  datadog_tracking_http_client: 2.1.0
  datadog_webview_tracking: 2.0.1
```

Build, Install & Run app

Observe crash when DD SDK is initilised

### Volume

All new builds

### Affected SDK versions

  datadog_flutter_plugin: 2.2.0   datadog_tracking_http_client: 2.1.0   datadog_webview_tracking: 2.0.1

### Does the crash manifest in the latest SDK version?

Yes

### Flutter Version

environment:   sdk: ^3.3.0   flutter: ^3.19.1

### Setup Type

Flutter Application

### Device Information

Any android device.

### Other relevant information

Probable root cause

Native SDK 2.7.0 is released yesterday, upon investigation I've found that we are not enforcing native SDK dependency versions. As `DefaultAppStartTimeProvider` is removed in Android SDK version 2.7.0 ([link](https://github.com/DataDog/dd-sdk-android/compare/2.5.1...2.7.0#diff-dc363543abff4764030144068cd9dd7411c50bf84c5fedefb52b903094d7d401L3)); app started crashing

Here is the Gradle dependency graph resolution

```
+--- project :datadog_webview_tracking
|    +--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.9.0 (*)
|    +--- io.flutter:flutter_embedding_debug:1.0.0-04817c99c9fd4956f27505204f7e344335810aed (*)
|    +--- com.datadoghq:dd-sdk-android-webview:2+ -> 2.7.0
|    |    \--- com.datadoghq:dd-sdk-android-core:2.7.0
|    |         +--- org.jetbrains.kotlin:kotlin-stdlib:1.8.10 -> 1.9.0 (*)
|    |         +--- com.squareup.okhttp3:okhttp:4.11.0 (*)
|    |         +--- com.google.code.gson:gson:2.10.1
|    |         +--- com.lyft.kronos:kronos-android:0.0.1-alpha11
|    |         |    +--- com.lyft.kronos:kronos-java:0.0.1-alpha11
|    |         |    |    \--- org.jetbrains.kotlin:kotlin-stdlib:1.4.10 -> 1.9.0 (*)
|    |         |    \--- org.jetbrains.kotlin:kotlin-stdlib:1.4.10 -> 1.9.0 (*)
|    |         +--- androidx.annotation:annotation:1.1.0 -> 1.7.1 (*)
|    |         +--- androidx.collection:collection:1.1.0 (*)
|    |         \--- androidx.work:work-runtime:2.8.1
|    |              +--- androidx.annotation:annotation-experimental:1.0.0 -> 1.3.1 (*)
|    |              +--- androidx.core:core:1.1.0 -> 1.12.0 (*)
|    |              +--- androidx.core:core:1.6.0 -> 1.12.0 (*)
|    |              +--- androidx.lifecycle:lifecycle-livedata:2.1.0 -> 2.6.1 (*)
|    |              +--- androidx.lifecycle:lifecycle-service:2.1.0 -> 2.6.1 (*)
|    |              +--- androidx.room:room-runtime:2.5.0
|    |              |    +--- androidx.annotation:annotation-experimental:1.1.0 -> 1.3.1 (*)
|    |              |    +--- androidx.arch.core:core-runtime:2.0.1 -> 2.2.0 (*)
|    |              |    +--- androidx.room:room-common:2.5.0
|    |              |    |    +--- androidx.annotation:annotation:1.3.0 -> 1.7.1 (*)
|    |              |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.20 -> 1.9.0 (*)
|    |              |    +--- androidx.sqlite:sqlite:2.3.0
|    |              |    |    +--- androidx.annotation:annotation:1.0.0 -> 1.7.1 (*)
|    |              |    |    \--- org.jetbrains.kotlin:kotlin-stdlib:1.7.20 -> 1.9.0 (*)
|    |              |    \--- androidx.sqlite:sqlite-framework:2.3.0
|    |              |         +--- androidx.annotation:annotation:1.2.0 -> 1.7.1 (*)
|    |              |         +--- androidx.sqlite:sqlite:2.3.0 (*)
|    |              |         \--- org.jetbrains.kotlin:kotlin-stdlib:1.7.20 -> 1.9.0 (*)
|    |              +--- androidx.sqlite:sqlite-framework:2.3.0 (*)
|    |              +--- androidx.startup:startup-runtime:1.0.0 -> 1.1.1 (*)
|    |              +--- com.google.guava:listenablefuture:1.0 -> 9999.0-empty-to-avoid-conflict-with-guava
|    |              \--- org.jetbrains.kotlin:kotlin-stdlib:1.7.21 -> 1.9.0 (*)
|    +--- project :webview_flutter_android (*)
|    \--- project :datadog_flutter_plugin
|         +--- com.datadoghq:dd-sdk-android-rum:2.5.0
|         |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.8.10 -> 1.9.0 (*)
|         |    +--- com.google.code.gson:gson:2.10.1
|         |    +--- com.squareup.okhttp3:okhttp:4.11.0 (*)
|         |    +--- androidx.core:core:1.3.1 -> 1.12.0 (*)
|         |    +--- androidx.metrics:metrics-performance:1.0.0-alpha04
|         |    |    +--- androidx.collection:collection:1.1.0 (*)
|         |    |    +--- androidx.core:core:1.5.0-beta01 -> 1.12.0 (*)
|         |    |    \--- org.jetbrains.kotlin:kotlin-stdlib:1.8.10 -> 1.9.0 (*)
|         |    +--- androidx.navigation:navigation-fragment:2.3.0 -> 2.6.0 (*)
|         |    +--- androidx.navigation:navigation-runtime:2.3.0 -> 2.6.0 (*)
|         |    +--- androidx.recyclerview:recyclerview:1.1.0 -> 1.3.0 (*)
|         |    \--- com.datadoghq:dd-sdk-android-core:2.5.0 -> 2.7.0 (*)
|         +--- com.datadoghq:dd-sdk-android-logs:2.5.0
|         |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.8.10 -> 1.9.0 (*)
|         |    +--- com.google.code.gson:gson:2.10.1
|         |    +--- androidx.annotation:annotation:1.1.0 -> 1.7.1 (*)
|         |    \--- com.datadoghq:dd-sdk-android-core:2.5.0 -> 2.7.0 (*)
|         +--- com.datadoghq:dd-sdk-android-ndk:2.5.0
|         |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.8.10 -> 1.9.0 (*)
|         |    +--- com.squareup.okhttp3:okhttp:4.11.0 (*)
|         |    +--- androidx.multidex:multidex:2.0.1
|         |    \--- com.datadoghq:dd-sdk-android-core:2.5.0 -> 2.7.0 (*)
|         +--- org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.8.22 -> 1.9.0 (*)
|         \--- com.google.code.gson:gson:2.8.8 -> 2.10.1
```

I do see a fix is commited in [here](https://github.com/DataDog/dd-sdk-flutter/compare/develop...release/datadog_flutter_plugin/v2.4.0) but haven't verified it yet.

Temporary fix applied in android/build.gradle

```
configurations.all { configuration ->
        configuration.resolutionStrategy.eachDependency { dependency ->
            if (dependency.requested.group == 'com.datadoghq') {
                // enforce DD dependency to 2.5.1
                dependency.useTarget "${dependency.requested.module}:2.5.1"
            }
        }
    }
```

**Comment by fuzzybinary** [comment]
Hey @sahildudeja-tide Thanks for raising the issue. It appears to be a bad constraint that slipped through on the webview plugin. I'll release a hotfix today.

**Comment by fuzzybinary** [comment]
We've restricted data `datadog_webview_tracking` to only support up to Native Android SDK version 2.5.1 (the last version the Core Datadog Flutter SDK shipped with) and published it as hot fix version 2.0.2.

I'm looking for longer term solution for our next release but this should resolve the immediate need.

Sorry for the inconvenience, and let me know if you see any issues with the fix.
